### ISSUE TITLE
Manual Input on Critical Archetype Cards

### ISSUE THREAD

**Issue by Lego6245** [jceddy/DailyArenaDeckAdvisor#26]
I'm unsure of the difficulty of this, but here's going.

There are some cards in certain decks that cannot be replaced, or rather, if they are replaced, drastically lower the power level of a deck. Replacing a lighting strike with a shock or a lava coil or a Jaya's greeting or a... something that deals some damage isn't the biggest problem, often times a deck will function (to varying degrees) without it. But say a person is making a Cavalcade deck and they lack... 3-4 copies of Cavalcade. In that case, the deck will simply not function as designed, and will be drastically worse than other possibilities.

The recommend does a good faith effort to find similar replacements where available, but for these sorts of cards (the one where their exclusion either changes the archetype or causes the win-rate to plummet.

This is somewhat similar to #8 but focuses specifically on the case of "these suggestions are so bad they drastically altar the power level of my deck"

**Comment by Lego6245** [comment]
The ideal end state is these decks are ranked lower, or these replacements are highlighted as "craft first before playing". Perhaps the later might be best, unless the replacement recommendation ranks the cards in order of "criticalness" to a deck?

**Comment by jceddy** [comment]
So I think this is possible, but requires manual intervention that isn't happening now. Currently these decks are pulled off the 'net in an automated fashion and there isn't a great way for the algorithm to know that specific cards are "key" without some human telling it that it is.

For normal power differences that *is* detectible to some extent, since all of the recommendations are based on power level scores that ultimately come from humans, augmented by an adjustment that is calculated based on how often different cards show up together in different builds. This modifier is always going to do a better job when it has more deck data to pull from, which means in a smaller meta right after rotation it's not going to do the job as well as it would just before rotation happens when there is a ton of data available.

So...I think a real solution for this would require some user input to the system going back to the server (something this program hasn't done so far) with some algorithm to crunch through the user input data and distribute it out one form that can be used when generating repacemtn suggestions.

Definitely doable, but requiring a significant amount of work. I will think about this...it probably won't happen soon, bit eventually I think we can get some improvement I this area.

**Comment by Lego6245** [comment]
Yeah, definitely not saying that this is easy work, far from it. It's definitely a good long term goal to have, though :)

**Comment by jceddy** [comment]
I changed the name of this to better reflect how I think it will ultimately be handled.
